py-polars/polars/expr/expr.py

```diff
@@ -3905,8 +3915,52 @@ def filter(self, predicate: Expr) -> Self:
         │ g2        ┆ 0   ┆ 3   │
         └───────────┴─────┴─────┘
 
+        Filter expressions can also take constraints as keyword arguments.
+
+        >>> import polars.selectors as cs
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "key": ["a", "a", "a", "a", "b", "b", "b", "b", "b"],
+        ...         "n": [1, 2, 2, 3, 1, 3, 3, 2, 3],
+        ...     },
+        ... )
+        >>> df.group_by("key").agg(
+        ...     n_0=pl.col("n").filter(n=1).sum(),
+        ...     n_1=pl.col("n").filter(n=2).sum(),
+        ...     n_2=pl.col("n").filter(n=3).sum(),
+        ... ).sort(by="key")
```

Comment:
Very minor comment, but can you change the column names to match the filter?
```suggestion
        >>> df.group_by("key").agg(
        ...     n_1=pl.col("n").filter(n=1).sum(),
        ...     n_2=pl.col("n").filter(n=2).sum(),
        ...     n_3=pl.col("n").filter(n=3).sum(),
        ... ).sort(by="key")
```